src/Components/Fsi.fs

```diff
@@ -105,8 +105,10 @@ module Fsi =
     let private sendLastSelection () =
         match lastSelectionSent with
         | Some x ->
-            send x
-            |> Promise.suppress // prevent unhandled promise exception
+            if "FSharp.saveOnSendLastSelection" |> Configuration.get false then
+                commands.executeCommand "workbench.action.files.save" |> ignore
```

Comment:
`commands.executeCommand` (should) returns Promise. So you can do `|> Promise.bind(fun _ -> send x)`